py/server/deephaven/updateby.py

```diff
@@ -267,3 +266,85 @@ def forward_fill(cols: Union[str, List[str]]) -> UpdateByOperation:
         return UpdateByOperation(j_updateby_op=_JUpdateByOperation.Fill(cols))
     except Exception as e:
         raise DHError(e, "failed to create a forward fill UpdateByOperation.") from e
+
+
```

Comment:
I can see the benefits of folding them into one function, but I prefer to have separate functions (especially when the parameter sets are quite different), as it is more explicit, cleaner, less chance for users to make mistake as far as arguments are concerned, also easier for us to document/maintain the code.

Reply:
I don't mind the explicitness of the `ticks`/`time` signatures but I have near-zero python experience in a professional role.

Reply:
I'm not sure the parameter sets need to be that different. If these were combined into one function, you could have the first argument `ts_col` be optional with a default value of `None`. The rest of the arguments could look like they do in the time-based rolling sum functions, where each argument must be an int or string. In a tick case, you give ints. In the time case, if you give an int, it assumes seconds, and if you give a string, it can be whatever.

Then, the try/except block at the end can be separated into two parts, which depend on whether or not a timestamp column is given.

I think the benefits of combining each into one function outweigh the drawbacks.

Reply:
Essentially, for the rolling sum, it would be (without comments):

```python
def rolling_sum(ts_col: str = None, cols: Union[str, List[str]], rev_time: Union[int, str],
                     fwd_time: Union[int, str] = 0) -> UpdateByOperation:
    if ts_col:
        try:
            cols = to_sequence(cols)
            rev_time = _JDateTimeUtils.expressionToNanos(rev_time) if isinstance(rev_time, str) else rev_time
            fwd_time = _JDateTimeUtils.expressionToNanos(fwd_time) if isinstance(fwd_time, str) else fwd_time
            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingSum(ts_col, rev_time, fwd_time, *cols))
        except Exception as e:
            raise DHError(e, "failed to create a rolling sum (time) UpdateByOperation.") from e
    else:
        try:
            cols = to_sequence(cols)
            return UpdateByOperation(j_updateby_op=_JUpdateByOperation.RollingSum(rev_ticks, fwd_ticks, *cols))
        except Exception as e:
            raise DHError(e, "failed to create a rolling sum (tick) UpdateByOperation.") from e
```

Reply:
@jjbrosnan some extra testing in the `ticks` case will be needed. Time strings need to be rejected as they make no sense for row-based computations.

Reply:
Yes, it is doable (we'll need to add checks for invalid argument combinations) but to me the extra cost and the potential cognitive burden on a user to understand the difference outweighs the gains.

Reply:
Also having ts_col as optional pretty much forces the users to use keyword arguments in the tick case, which may not be a bad thing but then it is not consistent with the time case.